lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -351,20 +351,81 @@ test.serial('migrateGranuleRecord handles nullable fields on source granule data
   );
 });
 
-test.serial('migrateGranuleRecord throws RecordAlreadyMigrated error for already migrated record', async (t) => {
+test.serial('migrateGranuleRecord throws RecordAlreadyMigrated error if previously migrated record is newer', async (t) => {
   const {
     knex,
     testGranule,
   } = t.context;
 
-  await migrateGranuleRecord(testGranule, knex);
+  const testGranule1 = testGranule;
+  const testGranule2 = {
+    ...testGranule1,
+    updatedAt: Date.now() - 1000,
+  };
+
+  await migrateGranuleRecord(testGranule1, knex);
 
   await t.throwsAsync(
-    migrateGranuleRecord(testGranule, knex),
+    migrateGranuleRecord(testGranule2, knex),
     { instanceOf: RecordAlreadyMigrated }
   );
 });
 
+test.serial('migrateGranuleRecord throws error if upsert does not return any rows', async (t) => {
+  const {
+    knex,
+    testCollection,
+    testExecution,
+  } = t.context;
+
+  const testGranule = generateTestGranule({
+    collectionId: buildCollectionId(testCollection.name, testCollection.version),
+    execution: testExecution.url,
+  });
+
+  await migrateGranuleRecord(testGranule, knex);
+
+  const newerGranule = {
+    ...testGranule,
+    updatedAt: Date.now(),
+  };
+
+  await t.throwsAsync(
+    migrateGranuleRecord(newerGranule, knex),
```

Comment:
So we expect the upsert to affect no rows because `newerGranule` has a later `updatedAt` timestamp? I can't totally follow the test setup and how we guarantee the expected result

Reply:
Yeah, this test wasn't clear.

I've updated it explicitly set the `status` of the test granule. That's what will cause the upsert to return 0 rows and the migration to fail. Also added some commentary:

https://github.com/nasa/cumulus/pull/2162/commits/0f9cfdbcfd7fa541920454abeb4ff537917a79e5

Reply:
Update: changed the wording slightly and added a similar explanation to the PDR migration test:

https://github.com/nasa/cumulus/pull/2162/commits/ea03d80a16a603b86afc5c4b484f2e56b6072881